src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -307,6 +332,96 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             logger.Info(ErrorCode.SiloInitializingFinished, "-------------- Started silo {0}, ConsistentHashCode {1:X} --------------", SiloAddress.ToLongString(), SiloAddress.GetConsistentHashCode());
         }
 
+        private IServiceCollection RegisterSystemTypes()
+        {
+            //
+            // Register the system classes and grains in this method.
+            //
+
+            IServiceCollection serviceCollection = new ServiceCollection();
+
+            serviceCollection.AddTransient<ManagementGrain>();
+            serviceCollection.AddTransient<GrainBasedMembershipTable>();
+            serviceCollection.AddTransient<GrainBasedReminderTable>();
+            serviceCollection.AddTransient<PubSubRendezvousGrain>();
+            serviceCollection.AddTransient<MemoryStorageGrain>();
+
+            return serviceCollection;
+        }
+
+        private void HandleStartup(IServiceCollection serviceCollection)
+        {
+            if (String.IsNullOrWhiteSpace(nodeConfig.StartupTypeName))
+            {
+                return;
+            }
+
+            var startupType = System.Type.GetType(nodeConfig.StartupTypeName);
+
+            if (startupType == null)
+            {
+                throw new InvalidOperationException(string.Format("Can not locate the type specified in the configuration file: '{0}'.", nodeConfig.StartupTypeName));
+            }
+
+            var servicesMethod = FindConfigureServicesDelegate(startupType);
```

Comment:
Just adding a quick note. If you check on ASP.Net OWIN implementation, the Startup.cs class can be any class, as long as the Startup attribute is tagging it. There are actually some conventions like:
1. Look for a class which has the Startup attribute
2. If none is found, look for a class names Startup
3. Call the ```void Configure(IAppBuilder app)``` method on that class

In other words, regardless of the discovery method(attribute or class name convention), the class must have:

1. A parameterless ctor
2. A non-static void method named Configure and that accept 1 single parameter as IAppBuilder type.

So, even without a interface, there is a "pseudo-contract" there.